Classes/Settings.cs

```diff
@@ -490,5 +481,32 @@ protected static void SetValue(string propertyName, string value)
                 }
             }
         }
+
+        private static string GetValuePostfix(Model.Programme progInfo, string name, string defaultValue)
+        {
+            string value = null;
+            if (progInfo != null)
+            {
+                value = GetValue(name + progInfo.Progid.ToString(CultureInfo.InvariantCulture));
```

Comment:
Using the name of a programme as a postfix, instead of the progid, however, doesn't look like a hack at all, I think. I don't mean to do that, but just for the argument. If you would have a table programme_settings, then progid would have a place. For the default-settings a zero-record could be used.

Adding the properties to the programmes-table, would ask for a zero-record also, if one would not want to store the same properties at several places. But such a zero-record would result in problems elsewhere, unlike such a use in programme_settings.

So far I have no idea what the perfect solution might be.

Concerning the GUI, my idea is to have a button for each of the three properties to be able to reset those to the default (resulting in deleting the records from settings). These buttons should be grayed out if the settings are the same as the default.

It's not obvious to me at this moment that a progid-column would make it much more easy to handle requested changes:
```sql
... where progid = 10 and property="SaveFolder"
```
or
```sql
... where property="SaveFolder10"
```
or
```sql
... where property="SaveFolder.Mike Tech Show Podcast"
```
Of course, something like the following is not possible without a progid-column, instead one has to loop through all progid's.
```sql
delete from settings where progid not null and property="SaveFolder"
```